api/v1alpha1/proposedcommit_types.go

```diff
@@ -40,36 +40,38 @@ type ProposedCommitSpec struct {
 	// +kubebuilder:validation:Required
 	ActiveBranch string `json:"activeBranch"`
 
-	CommitStatuses []CommitStatusProposedCommitSpec `json:"commitStatuses,omitempty"`
-}
+	// ActiveCommitStatuses what statuses are required for the active branch
+	// +kubebuilder:validation:Optional
+	ActiveCommitStatuses []CommitStatusSelector `json:"activeCommitStatuses"`
 
-type CommitStatusProposedCommitSpec struct {
-	// Key staging hydrated branch
-	// +kubebuilder:validation:Required
-	Key string `json:"key"`
+	// ProposedCommitStatuses what statuses are required for the proposed branch
+	// +kubebuilder:validation:Optional
+	ProposedCommitStatuses []CommitStatusSelector `json:"proposedCommitStatuses"`
 }
 
-type CommitStatusProposedCommitStatus struct {
+type ProposedCommitCommitStatusState struct {
 	// Key staging hydrated branch
 	// +kubebuilder:validation:Required
 	Key string `json:"key"`
 
 	// Phase what phase is the status in
 	// +kubebuilder:validation:Required
-	Phase string `json:"phase"`
+	Status string `json:"status"`
```

Comment:
This is a finite list of possible states, right? Should it be typed? Should we have enum validation?